apps/examples/touchscreen/touchscreen_main.c

```diff
@@ -0,0 +1,207 @@
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/prctl.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <errno.h>
+#if !defined(CONFIG_DISABLE_POLL)
+#include <poll.h>
+#endif
+
+#include <tinyara/input/touchscreen.h>
+
+#if defined(CONFIG_TOUCH_POLL)
+static bool g_terminated;
+#endif
+
+#if defined(CONFIG_TOUCH_CALLBACK) 
+static struct touch_set_callback_s touch_set_callback;
+#endif
+
+#if defined(CONFIG_TOUCH)
+#if defined(CONFIG_TOUCH_POLL)
+static void touch_test(void)
+{
+	/* read first 10 events */
+	int ret;
+
+	int fd = open(TOUCH_DEV_PATH, O_RDONLY);
+	if (fd < 0) {
+		printf("Error: Failed to open /dev/input0, errno : %d\n", get_errno());
+		return;
+	}
+
+	struct pollfd fds[1];
+	fds[0].fd = fd;
+	fds[0].events = POLLIN;
+
+	struct touch_sample_s buf;
+	while (!g_terminated) {
+		poll(fds, 1, 5000);
+		if (fds[0].revents & POLLIN) {
+			ret = read(fd, &buf, sizeof(struct touch_sample_s));
+			if (ret != - 1) {
+				printf("Total touch points %d\n", buf.npoints);
+				for (int i = 0; i < buf.npoints; i++) {
+					printf("coordinates id: %d, x : %d y : %d touch type: %d\n", buf.point[i].id, buf.point[i].x, buf.point[i].y, buf.point[i].flags);
+					if (buf.point[i].flags == TOUCH_DOWN) {
+						printf("Touch press event \n");
+					} else if (buf.point[i].flags == TOUCH_MOVE) {
+						printf("Touch hold/move event \n");
+					} else if (buf.point[i].flags == TOUCH_UP) {
+						printf("Touch release event \n");
+					}
+				}
+			}
+		}
+	}
+	close(fd);
+}
+#elif defined(CONFIG_TOUCH_CALLBACK)
+void is_touch_detected(int status)
+{
+	if (status == OK) {
+		printf("Total touch points %d\n", touch_set_callback.touch_points->npoints);
+		for (int i = 0; i < touch_set_callback.touch_points->npoints; i++) {
+			printf("TOUCH COORDINATES id: %d, x : %d y : %d touch type: %d\n", touch_set_callback.touch_points->point[i].id, touch_set_callback.touch_points->point[i].x, touch_set_callback.touch_points->point[i].y, touch_set_callback.touch_points->point[i].flags);
+			if (touch_set_callback.touch_points->point[i].flags == TOUCH_DOWN) {
+				printf("Touch press event \n");
+			} else if (touch_set_callback.touch_points->point[i].flags == TOUCH_MOVE) {
+				printf("Touch hold/move event \n");
+			} else if (touch_set_callback.touch_points->point[i].flags == TOUCH_UP) {
+				printf("Touch release event \n");
+			}
+		}
+	} else if (status == -EINVAL) {	// -EINVAL Error signifies touch_set_callback.touch_points is NULL
+		if (touch_set_callback.touch_points == NULL) {
+			touch_set_callback.touch_points = (struct touch_sample_s *)malloc(sizeof(struct touch_sample_s));
+			if (!touch_set_callback.touch_points) {
+				printf("Error: Touch point buffer memory allocation failed\n");
+			}
+		}
+	} else {
+		printf("Error: Touch detection failed\n");
+	}
+}
+#endif
+#endif
+
+static int touchsceen_test_start(void)
+{
+	printf("touchscreen test start\n");
+#if defined(CONFIG_TOUCH)
```

Comment:
This is the touch example. Is this necessary to check here? I think we should make a dependency in Kconfig.